azurelinuxagent/common/osutil/default.py

```diff
@@ -682,23 +682,13 @@ def get_first_if(self):
 
         # for 64bit the size is 40 bytes
         # for 32bit the size is 32 bytes
-        python_arc = platform.architecture()[0]
-        struct_size = 32 if python_arc == '32bit' else 40
+        python_arch = platform.architecture()[0]
+        struct_size = 32 if python_arch == '32bit' else 40
 
-        sock = socket.socket(socket.AF_INET,
-                             socket.SOCK_DGRAM,
-                             socket.IPPROTO_UDP)
-        buff=array.array('B', b'\0' * (expected * struct_size))
-        param = struct.pack('iL',
-                            expected*struct_size,
-                            buff.buffer_info()[0])
-        ret = fcntl.ioctl(sock.fileno(), 0x8912, param)
-        retsize=(struct.unpack('iL', ret)[0])
-        if retsize == (expected * struct_size):
-            logger.warn(('SIOCGIFCONF returned more than {0} up '
-                         'network interfaces.'), expected)
-        sock = buff.tostring()
-        primary = bytearray(self.get_primary_interface(), encoding='utf-8')
+        sock = self._get_interface_list_as_string(expected, struct_size)
+
+        primary_intf = self.get_primary_interface()
+        primary = bytearray(primary_intf, encoding='utf-8')
         for i in range(0, struct_size * expected, struct_size):
             iface=sock[i:i+16].split(b'\0', 1)[0]
```

Comment:
I would feel a little better if you defined an IFNAMSIZ constant instead of hardcoding that 16.